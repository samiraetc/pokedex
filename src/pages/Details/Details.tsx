import React, {useEffect, useState} from 'react';
import { useParams } from 'react-router';
import { getPokemonData } from '../../api/pokeapi';
import Card from '../../components/Card/Card';
import Loading from '../../components/Loading/Loading';
import PokemonDetails from '../../components/PokemonDetails/PokemonDetails';

export const COMPONENT_ID = 'pokedex-details'

const Details = () => {
    const {id} = useParams();
    const [pokemons, setPokemons] = useState([] as any);
    const [loading, setLoading] = useState(true);


    const fetchPokemons = async () => {
        try {
            const data = await getPokemonData(id);
            setPokemons(data);
            setLoading(false);
        } catch (error) {
            console.log('fetchPokemons error:', error)
        }
    }

    useEffect(() => {
        fetchPokemons();
      }, [])


    return (
        <h1 id={COMPONENT_ID}>
          { !loading ? (<PokemonDetails pokemon={pokemons} />) : (<Loading />)}
        </h1>
    )
}

export default Details;