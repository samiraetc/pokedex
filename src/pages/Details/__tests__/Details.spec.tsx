import { render, configure, waitFor, getByText } from '@testing-library/react'
import { act } from 'react-dom/test-utils';
import Details, { COMPONENT_ID } from '../Details';

configure({
    testIdAttribute: 'id',
  });
  
describe('[DETAILS]', () => {
    test('should render Details Page', async () => {
        const {asFragment, getByTestId} =  render(<Details/>);
        expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

});