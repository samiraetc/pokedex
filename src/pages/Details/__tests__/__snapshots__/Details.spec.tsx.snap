// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[DETAILS] should render Details Page 1`] = `
<DocumentFragment>
  <h1
    id="pokedex-details"
  >
    <div
      id="pokedex-loading"
    >
      <img
        alt="pokebola icon"
        class="loading"
        src="default"
      />
    </div>
  </h1>
</DocumentFragment>
`;
